src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1167,30 +1135,55 @@ function _combine(f::AbstractVector{<:Pair},
                     if isnothing(agg2idx_map)
                         agg2idx_map = _agg2idx_map_helper(idx, idx_agg)
                     end
-                    res[i] = idx, res[i][2][agg2idx_map]
+                    res[i] = idx_agg, res[i][2][agg2idx_map]
                 elseif idx != res[i][1]
-                    throw(ArgumentError("all functions must return vectors of the same length"))
+                    if keeprows
+                        throw(ArgumentError("all functions must return vectors of " *
+                                            "the length equal to the group rows count " *
+                                            "in the source GroupedDataFrame"))
+                    else
+                        throw(ArgumentError("all functions must return vectors of the same length"))
+                    end
                 end
             end
         end
     end
+
+    # remember that here first field in res[i] is not useful - it is just needed
+    # to keep track how the column was generated
+    # a correct index is stored in idx variable
```

Comment:
Is that field never useful at all, or just for the loop below? It sounds weird to say we store a field that's actually not needed in that form.

Reply:
Well - it was in the original code (and it was not used). I have corrected the comment.

Now we use it - it tells us how the column was generated as we have three modes:
* from aggregate
* from a function
* by a direct reuse of the column from the parent (this happens in `select` and `transform` which keep the order of rows from the parent)